text/0093-metric-instruments.md

```diff
@@ -0,0 +1,104 @@
+# List of new metric instruments
+
+Formalize the new metric instruments proposed in [OTEP 88](https://github.com/open-telemetry/oteps/pull/88).
+
+## Motivation
+
+OTEP 88 introduced a framework for reasoning about new metric
+instruments with various refinements and ended with a [sample
+proposal](https://github.com/open-telemetry/oteps/pull/88#sample-proposal).
+This proposal uses that proposal as a starting point.
+
+## Explanation
+
+The four instrument refinements discussed in OTEP 88 are:
+
+* Sum-only: When computing only a sum is the instrument's primary purpose
+* Non-negative: When negative values are invalid
+* Precomputed-sum: When the application has computed a cumulative sum itself
+* Non-negative-rate: When a negative rate is invalid.
+
+OTEP 88 describes how we are meant to compute rate information from
+instruments having the Non-negative-rate refinement.  Temporal
+aggregation (over time) must be treated as a special case, compared
+with spatial aggregation (over labels).  The logic for computing rates
+depends on whether the Precomputed-sum refinement is present or not,
+which determines whether Delta or Cumulative values are being
+captured.
+
+OTEP 88 proposes that when adding new instruments, we specify
+instruments having a single purpose, each with a distinct set of
+refinements, and each with a carefully selected name for the
+properties of the instrument.
+
+OTEP 88 also also proposes to support language-specific specialization
+as well, to support built-in value types (e.g., timestamps).
+
+## Internal details
+
+The existing specification includes three instruments.  In this
+proposal, the two foundational instruments become abstract, in the
+sense that all synchronous instruments are instances of a Measure
+instrument and all asynchronous instruments are instances of an
+Observer instrument.  New instrument names are given to the
+unrestricted, unrefined versions of rthe foundational instruments:
+
+1. **Distribution** is an an unrefined Measure instrument.  Distribution accepts positive or negative values and uses MinMaxSumCount aggregation by default.
+2. **LastValueObserver** is an unrefined Observer instrument.  LastValueObserver accepts positive or negative values and uses MinMaxSumCount aggregation by default.
+
+The existing Counter instrument is unchanged in this proposal.  It has
+Sum-only and Non-negative refinements, which imply the
+Non-negative-rate refinement.  Counter uses Sum aggregation by default
+
+Two new synchronous instruments are introduced in this proposal.
+
+1. **UpDownCounter** is a Sum-only instrument with no other refinements.  It supports capturing positive and negative changes to a sum (deltas).  UpDownCounter uses Sum aggregation by default.
+2. **Timing** is a Non-negative instrument specialized for the native timestamp of the platform.  It ensures that values are always reported in conventional units (milliseconds).
```

Comment:
Is it worth calling out float vs integer on this one? 
Is the idea that OTel will say "this is the timing instrument, it will always measure times in milliseconds", or would the units be customizable by instrumentation?

Reply:
Personally I would do ns because that is usually the unit for the monotonic time available in couple of languages like Java, go. If we go with millis then is definitely a float :)

Reply:
for sure. I'm mostly wondering if we're planning on building that in to the instrument refinement. I think it's worth calling out, either way.

Reply:
It's surprising to see restrictions on the type (int/float) and unit (ns/ms) here when the refinements above don't mention either. I think it's fine for the spec to describe conventions for instruments, but this does suggest there are more refinements in practice than are described in this OTEP.